packages/blocks/plus_minus/src/index.js

```diff
@@ -0,0 +1,1048 @@
+    const argId = Blockly.utils.genUid();
+
+    this.addVarInput_(name, argId);
+    if (this.getInput('STACK')) {
+      this.moveInputBefore(argId, 'STACK');
+    } else {
+      this.moveInputBefore(argId, 'RETURN');
+    }
+
+    this.arguments_.push(name);
+    this.varIds_.push(variable.getId());
+    this.argumentVarModels_.push(variable);
+    this.argIds_.push(argId);
+  },
+
+  /**
+   * Removes the argument associated with the given argument ID from the block.
+   * @param {string} argId An ID used to track arguments on the block.
+   * @private
+   */
+  removeArg_: function(argId) {
+    if (!this.getInput(argId)) {
+      return;
+    }
+    // TODO: My life would be so much happier if this returned a boolean instead
+    //  of throwing an error. I'm saying at least 15% more joy.
+    this.removeInput(argId);
+    if (this.arguments_.length == 1) { // Becoming argumentless.
+      this.getInput('TOP').removeField('WITH');
+    }
+
+    const index = this.argIds_.indexOf(argId);
+    this.arguments_.splice(index, 1);
+    this.varIds_.splice(index, 1);
+    this.argumentVarModels_.splice(index, 1);
+    this.argIds_.splice(index, 1);
+  },
+
+  /**
+   * Appends the actual inputs and fields associated with an argument to the
+   * block.
+   * @param {string} name The name of the argument.
+   * @param {string} argId The UUID of the argument (different from var ID).
+   * @this Blockly.Block
+   * @private
+   */
+  addVarInput_: function(name, argId) {
+    const nameField = new Blockly.FieldTextInput(name, this.validator_);
+    nameField.onFinishEditing_ = this.finishEditing_.bind(nameField);
+    nameField.createdVarIds_ = [];
+
+    this.appendDummyInput(argId)
+        .setAlign(Blockly.ALIGN_RIGHT)
+        .appendField(new FieldMinus(argId))
+        .appendField('variable:') // Untranslated!
+        .appendField(nameField, argId); // The name of the field is the arg id.
+  },
+
+  /**
+   * Validates text entered into the argument name field.
+   * @param {string} newName The new text entered into the field.
+   * @return {?string} The field's new value.
+   * @this Blockly.FieldTextInput
+   */
+  validator_: function(newName) {
+    const sourceBlock = this.getSourceBlock();
+    newName = newName.replace(/[\s\xa0]+/g, ' ').replace(/^ | $/g, '');
+    if (!newName) {
+      return null;
+    }
+
+    // The field name (aka id) is always equal to the arg id.
+    const index = sourceBlock.argIds_.indexOf(this.name);
+    const caselessName = newName.toLowerCase();
+    for (let i = 0, name; (name = sourceBlock.arguments_[i]); i++) {
+      if (i == index) {
+        // Don't check self because if we just added whitespace this breaks.
+        continue;
+      }
+      if (caselessName == name.toLowerCase()) {
+        return null; // It matches, so it is invalid.
+      }
+    }
+
+    sourceBlock.arguments_[index] = newName;
+
+    // TODO: Maybe delete the pre-edit variable if it has no other uses.
```

Comment:
You can still get the variable with `workspace.getVariableById`

Reply:
oh sweet you're right! Do you want me to add this? The behavior will be different than core, but I think it will be better, and it won't affect serialization or anything.

Reply:
Sure, sounds good.